mithril-common/src/entities/block_range.rs

```diff
@@ -53,6 +53,21 @@ impl BlockRange {
         }
     }
 
+    /// Get the start of the block range that contains the given block number
+    pub fn start(number: BlockNumber) -> BlockNumber {
+        Self::start_with_length(number, Self::LENGTH)
+    }
+
+    /// Get all [BlockRange] contained in the given interval
+    pub fn all_ranges_in(interval: Range<BlockNumber>) -> Vec<BlockRange> {
```

Comment:
This does not look like an efficient implementation with millions of records.
You'd better use basic arithmetic to do the computation instead.